go-controller/pkg/libovsdbops/chassis.go

```diff
@@ -115,3 +140,142 @@ func DeleteChassisWithPredicate(sbClient libovsdbclient.Client, p chassisPredica
 	err := m.Delete(opModels...)
 	return err
 }
+
+// CreateOrUpdateRemoteChassis creates a chassis record if not present. It also
+// creates the Encap record required when creating a chassis record.
+// If the chassis record already exists, it marks the chassis as a remote
+// by setting the options:is-remote=true.
+// Note: Normally ovn-controller creates a chassis record when it connects to
+// OVN Southbound db.  But with multiple zones deployment, ovn-controller will create the
+// chassis record in its own local zone Southbound db. ovn-k network controller
+// manager needs to create a chassis records for these remote nodes in its zone Southbound db.
+func CreateOrUpdateRemoteChassis(sbClient libovsdbclient.Client, hostname, name, ip string) error {
```

Comment:
We don't usually hardcode or check for specific data within the libovsdbops methods (which means that you are actually including bussiness logic in a simple library to operate against the DB). 

So you would pass in a sbdb.Chassis and a sbdb.Encap as you like to have them in the DB and have here pretty generic methods like `CreateOrUpdateChassis`, `DeleteChassis` and not specific Remote/Local/convert methods which would be a layer above this one.

This is a nit of course while this is the only use we have for it, but the implementation would have to be changed or duped if we had other reasons and data in the future to write to the SBDB chassis table.

This comment applies to the other methods here as well.

Reply:
yeah I had the same-ish comments...

Reply:
+1